File: debug_spell_data.py
#!/usr/bin/env python3
"""
Community Dragon API에서 오공 Q 스킬 데이터 구조 확인
"""

import json
import urllib.request

def fetch_json(url):
    with urllib.request.urlopen(url) as response:
        return json.loads(response.read())

def main():
    print("Community Dragon API에서 오공 데이터 가져오는 중...")
    data = fetch_json("https://raw.communitydragon.org/latest/game/data/characters/monkeyking/monkeyking.bin.json")
    
    # Q 스킬 관련 키 찾기
    q_keys = [k for k in data.keys() if "DoubleAttack" in k or "MonkeyKingQ" in k]
    print("\n=== Q 스킬 관련 키들 ===")
    for key in q_keys[:10]:
        print(key)
    
    # 메인 Q 스킬 데이터 확인
    main_q_key = "Characters/MonkeyKing/Spells/MonkeyKingDoubleAttackAbility/MonkeyKingDoubleAttack"
    if main_q_key in data:
        print("\n=== 메인 Q 스킬 데이터 구조 ===")
        q_data = data[main_q_key]
        print(f"Top level keys: {list(q_data.keys())[:10]}")
        
        if "mSpell" in q_data:
            m_spell = q_data["mSpell"]
            print("\n=== mSpell 키들 ===")
            spell_keys = list(m_spell.keys())[:30]
            print(spell_keys)
            
            # DataValues 확인
            if "DataValues" in m_spell:
                print("\n=== DataValues ===")
                data_values = m_spell["DataValues"]
                print(f"Type: {type(data_values)}")
                if isinstance(data_values, dict):
                    dv_keys = list(data_values.keys())[:20]
                    print(f"DataValues keys: {dv_keys}")
                    for key in dv_keys[:5]:
                        val = data_values[key]
                        val_str = json.dumps(val)[:100]
                        print(f"{key}: {type(val).__name__} = {val_str}")
                elif isinstance(data_values, list):
                    print(f"DataValues is a list with {len(data_values)} items")
                    print(f"First few items: {data_values[:5]}")
            
            # mSpellCalculations 확인
            if "mSpellCalculations" in m_spell:
                print("\n=== mSpellCalculations ===")
                calc = m_spell["mSpellCalculations"]
                if isinstance(calc, dict):
                    calc_keys = list(calc.keys())[:20]
                    print(f"Keys: {calc_keys}")
                    for key in calc_keys[:5]:
                        val = calc[key]
                        print(f"{key}: {type(val).__name__}")
                        if isinstance(val, dict):
                            print(f"  Sub-keys: {list(val.keys())[:5]}")
            
            # cooldownTime 확인
            if "cooldownTime" in m_spell:
                print("\n=== cooldownTime ===")
                print(m_spell["cooldownTime"])
            
            # armorshred 관련 키 찾기
            print("\n=== armorshred 관련 키 검색 ===")
            all_keys = []
            if isinstance(m_spell, dict):
                all_keys.extend(m_spell.keys())
                for key, val in m_spell.items():
                    if isinstance(val, dict):
                        all_keys.extend([f"{key}.{k}" for k in val.keys()])
            
            armor_keys = [k for k in all_keys if "armor" in k.lower() or "shred" in k.lower()]
            print(f"Found {len(armor_keys)} keys: {armor_keys[:10]}")

if __name__ == "__main__":
    main()

File: debug-spell-data.js
/**
 * Community Dragon API에서 오공 Q 스킬 데이터 구조 확인
 * Node.js로 실행: node debug-spell-data.js
 */

const https = require('https');

function fetchJSON(url) {
  return new Promise((resolve, reject) => {
    https.get(url, (res) => {
      let data = '';
      res.on('data', (chunk) => { data += chunk; });
      res.on('end', () => {
        try {
          resolve(JSON.parse(data));
        } catch (e) {
          reject(e);
        }
      });
    }).on('error', reject);
  });
}

async function debugSpellData() {
  try {
    console.log('Community Dragon API에서 오공 데이터 가져오는 중...');
    const data = await fetchJSON('https://raw.communitydragon.org/latest/game/data/characters/monkeyking/monkeyking.bin.json');
    
    // Q 스킬 관련 키 찾기
    const qKeys = Object.keys(data).filter(k => k.includes('DoubleAttack') || k.includes('MonkeyKingQ'));
    console.log('\n=== Q 스킬 관련 키들 ===');
    qKeys.forEach(key => console.log(key));
    
    // 메인 Q 스킬 데이터 확인
    const mainQKey = 'Characters/MonkeyKing/Spells/MonkeyKingDoubleAttackAbility/MonkeyKingDoubleAttack';
    if (data[mainQKey]) {
      console.log('\n=== 메인 Q 스킬 데이터 구조 ===');
      const qData = data[mainQKey];
      console.log('Top level keys:', Object.keys(qData));
      
      if (qData.mSpell) {
        const mSpell = qData.mSpell;
        console.log('\n=== mSpell 키들 ===');
        console.log(Object.keys(mSpell).slice(0, 30));
        
        // DataValues 확인
        if (mSpell.DataValues) {
          console.log('\n=== DataValues ===');
          console.log('Type:', typeof mSpell.DataValues);
          if (typeof mSpell.DataValues === 'object' && !Array.isArray(mSpell.DataValues)) {
            const dvKeys = Object.keys(mSpell.DataValues);
            console.log('DataValues keys:', dvKeys.slice(0, 20));
            dvKeys.slice(0, 5).forEach(key => {
              const val = mSpell.DataValues[key];
              console.log(`${key}: ${typeof val} = ${JSON.stringify(val).substring(0, 100)}`);
            });
          }
        }
        
        // mSpellCalculations 확인
        if (mSpell.mSpellCalculations) {
          console.log('\n=== mSpellCalculations ===');
          const calcKeys = Object.keys(mSpell.mSpellCalculations);
          console.log('Keys:', calcKeys.slice(0, 20));
          calcKeys.slice(0, 5).forEach(key => {
            const val = mSpell.mSpellCalculations[key];
            console.log(`${key}: ${typeof val}`);
            if (typeof val === 'object' && val !== null) {
              console.log(`  Sub-keys: ${Object.keys(val).slice(0, 5).join(', ')}`);
            }
          });
        }
        
        // cooldownTime 확인
        if (mSpell.cooldownTime) {
          console.log('\n=== cooldownTime ===');
          console.log(mSpell.cooldownTime);
        }
      }
    }
    
    // 다른 관련 스킬 데이터도 확인
    console.log('\n=== 다른 관련 스킬 데이터 확인 ===');
    qKeys.slice(0, 3).forEach(key => {
      const spellData = data[key];
      if (spellData?.mSpell) {
        console.log(`\n${key}:`);
        const mSpell = spellData.mSpell;
        if (mSpell.DataValues) {
          console.log('  DataValues keys:', Object.keys(mSpell.DataValues).slice(0, 10));
        }
      }
    });
    
  } catch (error) {
    console.error('Error:', error);
  }
}

debugSpellData();


